atc/engine/builder/step_factory.go

```diff
@@ -89,7 +86,7 @@ func (factory *stepFactory) PutStep(
 		factory.resourceFactory,
 		factory.resourceConfigFactory,
 		factory.strategy,
-		factory.pool,
+		factory.client,
```

Comment:
What's the context behind removing pool from PutStep only but not for the others?

Reply:
good catch